gdx/src/com/badlogic/gdx/utils/ObjectMap.java

```diff
@@ -1,846 +1,782 @@
+ * for more information on Fibonacci hashing. It uses linear probing to resolve collisions, which is far from the academically
+ * optimal algorithm, but performs considerably better in practice than most alternatives, and combined with Fibonacci hashing, it
+ * can handle "normal" generated hashCode() implementations, and not just theoretically optimal hashing functions. Even if all
+ * hashCode()s this is given collide, it will still work, just slowly; the older libGDX implementation using cuckoo hashing would
+ * crash with an OutOfMemoryError with under 50 collisions.
+ * <br>
+ * This map performs very fast contains and remove (typically O(1), worst case O(n) due to occasional probing, but still very
+ * fast). Add may be a bit slower, depending on hash collisions, but this data structure is somewhat collision-resistant.
+ * Load factors greater than 0.91 greatly increase the chances the map will have to rehash to the next higher POT size.
+ * Memory usage is excellent, and the aforementioned collision-resistance helps avoid too much capacity resizing.
+ * <br>
+* Iteration should be fast with OrderedSet and OrderedMap, whereas ObjectSet and ObjectMap aren't designed to provide especially
+ * quick iteration.
+ *
+ * @author Tommy Ettinger
+ * @author Nathan Sweet
+ */
+public class ObjectMap<K, V> implements Iterable<ObjectMap.Entry<K, V>> {
+	static final Object dummy = new Object();
+
+	public int size;
+
+	K[] keyTable;
+	V[] valueTable;
+
+	float loadFactor;
+	int threshold;
+	/**
+	 * Used by {@link #place(Object)} to bit-shift the upper bits of a {@code long} into a usable range (less than or
+	 * equal to {@link #mask}, greater than or equal to 0). If you're setting it in a subclass, this shift can be
+	 * negative, which is a convenient way to match the number of bits in mask; if mask is a 7-bit number, then a shift
+	 * of -7 will correctly shift the upper 7 bits into the lowest 7 positions. If using what this class sets, shift
+	 * will be greater than 32 and less than 64; if you use this shift with an int, it will still correctly move the
+	 * upper bits of an int to the lower bits, thanks to Java's implicit modulus on shifts.
+	 * <br>
+	 * You can also use {@link #mask} to mask the low bits of a number, which may be faster for some hashCode()s, if you
+	 * reimplement {@link #place(Object)}.
+	 */
+	protected int shift;
+	/**
+	 * The bitmask used to contain hashCode()s to the indices that can be fit into the key array this uses. This should
+	 * always be all-1-bits in its low positions; that is, it must be a power of two minus 1. If you subclass and change
+	 * {@link #place(Object)}, you may want to use this instead of {@link #shift} to isolate usable bits of a hash.
+	 */
+	protected int mask;
+
+	Entries entries1, entries2;
+	Values values1, values2;
+	Keys keys1, keys2;
+
+	/**
+	 * Creates a new map with an initial capacity of 51 and a load factor of 0.8.
+	 */
```

Comment:
Should run the libgdx source code formatter so this matches the style of the rest of the source.

Reply:
I did, that's the only reason why it looks like this. Either the source code formatter as exactly posted in the libGDX repo is producing different results on my machine than yours, or perhaps you never ran it on the original ObjectMap over 5 years ago. I point to #5698 and gesture.

Reply:
For some reason your source does not have the right format. Which formatter did you use? [This](https://github.com/libgdx/libgdx/blob/master/eclipse-formatter.xml) is the one. I don't know if the IDEA formatter is equivalent. If it is not, it should be fixed.

All of the libgdx classes I maintain use the formatter. Other contributors tend to be less strict. You can test your formatter by choosing a class I maintain, eg `FloatArray`, run the formatter, and the result should be no changes.

Reply:
Yep, that's the one I used, and installed as the default in IntelliJ using the instructions on the official site. I'll download it again and make sure the formatter is exactly as-downloaded. If the linked Eclipse formatter doesn't produce the same results across different installations (especially since **the official site strongly recommends IDEA**), then I think a Gradle plugin is required if formatting somehow matters.

I also want to emphasize that libGDX's code is not a beauty pageant, and this is a bug fix.

Reply:
Yep, reinstalled eclipse-formatter.xml , it brings all sorts of new unpleasantness to `FloatArray`: https://gist.github.com/tommyettinger/9c43514f5b4a9d98675829f16862524c <-- here, Before is FloatArray from the repo, After is with eclipse-formatter run (check the indents, it mixes tab and spaces), and After_IDEA uses idea-formatter (which fixes the mixing, but changes comments and single-line if statements). I'm pretty sure I had to manually fix the spaces and tabs once, and I vaguely remember doing so in a past contribution, then I must have gotten it to match the repo behavior and use just tabs.

Spending this much time and effort on non-functionality-related code-style is silly. If eclipse-formatter.xml from *this repo* and idea-formatter.xml from *this repo* don't replicate the style you want, **if the style matters to you**, then add instructions or tools to *this repo or its associated documentation* to allow contributors to produce that style. If I can't produce this style automatically, I simply won't do it; the effort to hand-tweak every one of the many lines in this PR, to match a code style that the IDE will fight me on at every step, is absurd. If you can do it automatically, then do it. You, @NathanSweet , have the ability to edit this PR; only you, Mario Zechner, and I can do that to my knowledge. Of those three, I can't seem to get any IDE to reproduce the code style used in libGDX given the XML files in libGDX itself, Mario I don't think is still involved in libGDX (at least for PRs), and that leaves you.

Reply:
@tommyettinger is making a very valid point here. The current Eclipse formatter file produces different outputs depending on the IDE. If PRs are going to be rejected based on code format (this is not the first PR that is on hold for this reason) and there's no way to ensure a standard format across IDEs, formatting should be made automatic. Alternatively some flexibility should apply on the merging policy. Wasting time on manually editing lines to please a very specific non automatable code style test is quite demoralizing for contributors.

Reply:
IMO code style is important to normalize contributions. No one suggested manually matching the style. If the Eclipse formatter doesn't work as expected in IDEA and the IDEA formatter doesn't match Eclipse, then it would be helpful to find a solution. Maybe #5698 is that, I haven't had time to do more than read the issue.

@Tom-Ski added the IDEA formatter file, maybe he has some insight on why it doesn't match Eclipse or if it can be made to match?

Reply:
For this PR I'll format and merge.

Reply:
I pretty much only format method code on rare occasions, the rest is just the style that I am used to that matches. It _should_ in theory be ok because this comes directly from IDEA conversion. If it doesn't we should address it in a different issue and try to get these matched up. I agree with the general sentiments regarding this issue though, and only really have problems when code styles in the code don't match. Spaces around methods, wrapping, and indentation values.  These are the only critical things for me at least.

@tommyettinger just a side, the reason why that is recommended because it is the most robust way to get setup with the project and developing quickly for all platforms.